lib/request.js

```diff
@@ -1,31 +1,94 @@
 import got from 'got';
 import logger from './logger.js';
 import { extend } from './utils/index.js';
-
-function getMimeType (contentType) {
-	return contentType ? contentType.split(';')[0] : null;
-}
+import * as cheerio from 'cheerio';
 
 function defaultResponseHandler ({response}) {
 	return Promise.resolve(response);
 }
 
-function extractEncodingFromHeader (headers) {
-	const contentTypeHeader = headers['content-type'];
+function extractEncodingFromResponse (response) {
+	if (typeof response?.headers === 'object') {
+		const contentTypeHeader = response.headers['content-type'];
+
+		return contentTypeHeader && contentTypeHeader.includes('utf-8') ? 'utf8' : 'binary';
+	}
+
+	return undefined;
```

Comment:
why do we return `undefined` from this functions, but `null` from `extractMimeTypeFromResponse`? Maybe it will be better to return same resule (null or undefined) from both functions because they seem to be very similar.